pkg/controller/common/helpers.go

```diff
@@ -539,12 +551,44 @@ func ValidateMachineConfig(cfg mcfgv1.MachineConfigSpec) error {
 	return nil
 }
 
+// ExtractIgnitionKargsFor4_13 parses a raw ignition config to 3.4 and extracts the kernel args,
+// erroring if ShouldNotExist is populated (since this MCO doesn't know how to implement it). This
+// function is used by the render controller to extract the kargs from ignition and migrate them to
+// MachineConfig so newer versions that specify kargs in ignition will still work during a downgrade.
+// TODO(jkyros): remove this when we raise the ignition default to 3.4
+func ExtractIgnitionKargsFor4_13(rawIgn []byte) ([]ign3_4types.KernelArgument, error) {
+	ignCfgV3, rptV3, errV3 := ign3_4.ParseCompatibleVersion(rawIgn)
+	// No kargs in an empty config
+	if errors.Is(errV3, ign3errors.ErrEmpty) {
+		return nil, nil
+	}
+	if errV3 == nil && !rptV3.IsFatal() {
+		// We can't reconcile ShoultNotExist
+		if len(ignCfgV3.KernelArguments.ShouldNotExist) > 0 {
+			return nil, fmt.Errorf("Ignition KernelArguments.ShouldNotExist is not supported in this release ( ShouldNotExist: %s was supplied )", ignCfgV3.KernelArguments.ShouldNotExist)
+		}
+
+		// But we can stuff ShouldExist in MachineConfig, so send those back
```

Comment:
One note: Ignition `KernelArguments` are allowed to contain spaces, and this is encouraged when multiple arguments are order-dependent.  For example, `console=tty0 console=ttyS0,115200` might be written as a single KernelArgument.  The MCO karg code appears to always split on spaces, though.  Here we're only concerned with downgrades, so if the future MCO never creates these compound arguments when converting from MachineConfig, this should all be fine.